urllib3/util/selectors.py

```diff
@@ -133,13 +119,7 @@ def _syscall_wrapper(func, recalc_timeout, *args, **kwargs):
                         if current_time > expires:
                             raise OSError(errno.ETIMEDOUT, "Connection timed out")
                         if recalc_timeout:
-                            # FIXME: we're now not updating the timeout argument in kwargs
-                            #        (We never updated it in args)
-                            #        Note that we're not currently testing to see if
-                            #        a passed-in timeout is reaching the select.select
-                            #        (I strongly suspect it is not in some cases)
-                            if timeout is not None:
-                                timeout = expires - current_time
+                            kwargs["timeout"] = expires - current_time
```

Comment:
This should probably be `timeout = expires - current_time` so the value is updated to be used in the next attempt at the syscall.